src/app/action-items/action-items.component.ts

```diff
@@ -1,38 +1,66 @@
-import { ActionTasksElement } from './../services/tasksService/tasks.service';
+import { AddItemComponent } from './../add-item/add-item.component';
 import { Component, OnInit, OnDestroy } from '@angular/core';
 import { map } from 'rxjs/operators';
-import { TasksService } from '../services/tasksService/tasks.service';
+import { TasksService, ActionItem } from '../services/tasksService/tasks.service';
 import { DaysLeftToDeadlineService } from '../services/daysLeftToDeadlineService/days-left-to-deadline.service';
+import { MatDialog, MatDialogConfig } from '@angular/material/dialog';
+import { Subscription } from 'rxjs';
 
-export interface ActionTasksElementMapped extends ActionTasksElement {
-  dueDay: number;
+export interface ActionItemMapped extends ActionItem {
+  dueDay?: number;
 }
 
 @Component({
   selector: 'app-action-items',
   templateUrl: './action-items.component.html',
   styleUrls: ['./action-items.component.css']
 })
-export class ActionItemsComponent implements OnInit {
-  dataSource: ActionTasksElementMapped[];
-  loading = true;
-  currentDate = new Date();
-  constructor(private tasksService: TasksService, private daysCountService: DaysLeftToDeadlineService) {}
+export class ActionItemsComponent implements OnInit, OnDestroy {
+  dataSource: ActionItemMapped[];
+  isloadingActionItems = false;
+  private currentDate: Date = new Date();
+  private subscription: Subscription;
+  constructor(
+    private tasksService: TasksService,
+    private daysCountService: DaysLeftToDeadlineService,
+    private matDialog: MatDialog
+  ) {}
   ngOnInit() {
-    this.tasksService
-      .getAllTasks()
+    this.retrieveActionItems();
+  }
+  retrieveActionItems() {
+    this.isloadingActionItems = true;
+    this.subscription = this.tasksService
```

Comment:
**SUGGESTION:** As we discussed - if you used `first` operator then you wouldn't need this subscription variable, as it'd automatically unsubscribe after getting first set of data.